shared/src/metabase/shared/parameters/parameters.cljc

```diff
@@ -63,40 +63,59 @@
            (formatted-value :date/single end locale))
       "")))
 
+(defn- translated-interval
+  [interval n]
+  (get
+   {"minutes"  (trsn "Minute" "Minutes" n)
+    "hours"    (trsn "Hour" "Hours" n)
+    "days"     (trsn "Day" "Days" n)
+    "weeks"    (trsn "Week" "Weeks" n)
+    "months"   (trsn "Month" "Months" n)
+    "quarters" (trsn "Quarter" "Quarters" n)
+    "years"    (trsn "Year" "Years" n)}
+   interval))
+
+(defn- format-relative-date
+  [prefix n interval]
+  (let [n        #?(:clj (Integer. ^String n) :cljs (js/parseInt n))
+        interval (translated-interval interval n)]
+    (case [prefix (= n 1)]
```

Comment:
`interval` is translated using the `trsn` macro which correctly chooses the appropriate plural form of the noun in the target language for the value `n`.

This uses ttag's `ngettext` function on the frontend, and on the backend I added support for this in https://github.com/metabase/metabase/pull/24248

Reply:
I wasn't thinking about the implementation. I was discussing the requirements: if I understand correctly, the requirement is to be able to distinguish singular and plural, although there are languages which have other forms too, e.g., [dual](https://en.wikipedia.org/wiki/Dual_(grammatical_number)). I don't think you have to do anything with this, I was just curious how translations are supposed to work in such languages.

Reply:
Yeah, that's what `trsn` is for. It's already implemented -- see the PR I linked above for an explanation of how it works. The core code for the backend implementation is in `metabase.util.i18n.plural`.

Reply:
In this spot in particular, we're omitting the number when it is equal to 1. (i.e. `Previous Week` instead of `Previous 1 Week`). I grant that this specific construct might not translate well to other languages but it's what we're already doing when displaying parameter values in the widgets.

In general, translating a string that's constructed programmatically, and having it still be grammatical, is an unsolved problem in our codebase, unfortunately.